src/macros.jl

```diff
@@ -1005,34 +1005,178 @@ function parsevariable(_error::Function, infoexpr::VariableInfoExpr, lvalue, lsi
-#    * the `Symmetric` argument will check that the container is symmetric and only allocate one variable for each pair of non-diagonal entries.
-#    * the `PSD` argument will do the same as `Symmetric` but in addition it will specify that the variables created belongs to the PSD cone in the `varCones` field of the model.
-#   Moreover, Int and Bin are special keywords that are equivalent to `integer=true` and `binary=true`.
-# * The keyword arguments start, lowerbound, upperbound, binary, and integer category may not be passed as is to
-#   `buildvariable` since they may be altered by the parsing of `expr` and we may need to pass it pointwise if it is a container since
-#   `buildvariable` is called separately for each variable of the container. Moreover they will be passed inside the VariableInfo object.
-# * A custom error function is passed as positional argument to print the full @variable call before the error message.
-# * The keyword argument basename is not passed to `constructconstraint!` but rather to `addvariable`.
-#
-# Examples:
-# * `@variable(m, x >= 0)` is equivalent to `x = addvariable(m, buildvariable(msg -> error("In @variable(m, x >= 0): ", msg), VariableInfo(true, 0, false, NaN, false, NaN, false, NaN, false, false)), "x")`
-# * `@variable(m, x[1:N,1:N], Symmetric, Poly(X))` is equivalent to
-#   ```
-#   x = Matrix{...}(N, N)
-#   for i in 1:N
-#       for j in 1:N
-#           x[i,j] = x[j,i] = addvariable(m, buildvariable(msg -> error("In @variable(m, x[1:N,1:N], Symmetric, Poly(X)): ", msg), VariableInfo(false, NaN, false, NaN, false, NaN, false, NaN, false, false), Poly(X)), "")
-#       end
-#   end
-#   ```
+"""
+    @variable(m; kwargs...)
+
+Add an *anonymous* (see the "Variable naming" section below) variable to the
+model `m` described by the keyword arguments `kwargs` that belongs.
+
+    @variable(m, expr, args...; kwargs...)
+
+Add a variable to the model `m` described by the expression `expr`, the
+positional arguments `args` and the keyword arguments `kwargs` that belongs.
+The expression `expr` can either be (note that in the following the symbol
+`<=` can be used instead of `≤` and the symbol `>=`can be used instead of
+`≥`)
+
+* of the form `varexpr` creating variables described by `varexpr`;
+* of the form `varexpr ≤ ub` (resp. `varexpr ≥ lb`) creating variables described by
+  `varexpr` with upper bounds given by `ub` (resp. lower bounds given by `lb`);
+* of the form `varexpr == value` creating variables described by `varexpr` with
+  fixed values given by `value`;
+* of the form `lb ≤ varexpr ≤ ub` or `ub ≥ varexpr ≥ lb` creating variables
+  described by `varexpr` with lower bounds given by `lb` and upper bounds given
+  by `ub`.
+
+The expression `varexpr` can either be
+
+* of the form `varname` creating a scalar real variable of name `varname`;
+* of the form `varname[idxs1,idxs2,...,idxn]` creating an `n`-dimensional
+  `Array` or `JuMPArray` of scalar real variables.
```

Comment:
Nit: Here and throughout. I'm not sure what our style policy on lists is, but it either needs to be full sentences ending in ".", or fragments separated by ";", with the penultimate item having "; and" or "; or", and the final item ending in ".". At the very least, let's not mix styles in the same list.